### PR TITLE
Business

### PR DESCRIPTION
**Pull request by jazzmano**
alot of javadoc and unused imports

### CODE REVIEW

File: HospitalGame_Business/src/business/BusinessFacade.java [hunk 1]
```diff
@@ -87,7 +94,7 @@ public BusinessFacade() {
     private void createRooms(int numberOfRooms) {
```
Comment (frederikrosenberg) [inline HospitalGame_Business/src/business/BusinessFacade.java:105]: Javadoc missing

File: HospitalGame_Business/src/business/BusinessFacade.java [hunk 2]
```diff
@@ -129,11 +136,11 @@ private void createRooms(int numberOfRooms) {
...
         npcFacade.create(NPCID.PORTER, false, "porter");
         npcFacade.create(NPCID.COMPUTER, false, "computer");
-        
+
         // Sets the current room for the player, and generates the rooms.
         player.setCurrentRoom(map.generateMap(numberOfRooms, items, Arrays.asList(npcFacade.getNPCs())).getRoomID());
     }
-    
+
     private void reset() {
```
Comment (frederikrosenberg) [inline HospitalGame_Business/src/business/BusinessFacade.java]: Javadoc missing

File: HospitalGame_Business/src/business/BusinessFacade.java [hunk 3]
```diff
@@ -347,39 +359,65 @@ public String interact(IPlayer player, INPC npc) {
...
     @Override
     public boolean addHighScore(String name) {
         return highScore.addHighScore(name, score);
     }
 
+    /**
+     * if the highscore has changed, then save this highscore
+     */
     @Override
     public void closing() {
```
Comment (frederikrosenberg) [inline HospitalGame_Business/src/business/BusinessFacade.java]: Not correct.
This should be something like:
This function should be called before the game is closed

File: HospitalGame_Business/src/business/BusinessFacade.java [hunk 4]
```diff
@@ -347,39 +359,65 @@ public String interact(IPlayer player, INPC npc) {
...
+     */
     @Override
     public boolean isHighScoreNameTaken(String name) {
         return highScore.isNameTaken(name);
     }
 
+    /**
+     * checks if game is saveable
+     *
+     * @returns true if it can be saved
```
Comment (frederikrosenberg) [inline HospitalGame_Business/src/business/BusinessFacade.java]: No is checks if there is a saved game on the filesystem

File: HospitalGame_Business/src/business/BusinessHighScore.java [hunk 5]
```diff
@@ -10,67 +10,120 @@
...
     private final int store = 10;
-    
+
     /**
      * If the high scores has been updated
      */
     private boolean isDirty = false;
-    
+
     private HashMap<String, Integer> scores;
```
Comment (frederikrosenberg) [inline HospitalGame_Business/src/business/BusinessHighScore.java]: Javadoc missing

File: HospitalGame_Business/src/business/BusinessHighScore.java [hunk 6]
```diff
@@ -10,67 +10,120 @@
...
         return isDirty;
     }
 
+    /**
+     * checks if the score is eligible for a highscore
+     *
+     * @param score the score that represents a HighScore
+     * @return a place on the highscore
+     */
     public boolean eligibleForHighscore(int score) {
```
Comment (frederikrosenberg) [inline HospitalGame_Business/src/business/BusinessHighScore.java]: It does not return a place on the high score. It returns true if the score is high score is enough to get on the high score list

File: HospitalGame_Business/src/business/BusinessHighScore.java [hunk 7]
```diff
@@ -10,67 +10,120 @@
...
+        }
         return Collections.min(scores.values()) < score;
     }
-    
+
+    /**
+     * checks if the name can be used
+     *
+     * @param name of the player
+     * @return name in highscore
```
Comment (frederikrosenberg) [inline HospitalGame_Business/src/business/BusinessHighScore.java]: It does not return a name?
It returns true if the given name is taken

File: HospitalGame_Business/src/business/BusinessHighScore.java [hunk 8]
```diff
@@ -10,67 +10,120 @@
...
+
+    /**
+     * adds highscore if score and name is eligible and not taken
+     *
+     * @param name of the player
+     * @param score of the player
+     * @return adds a highscore
+     */
+
     public boolean addHighScore(String name, int score) {
```
Comment (frederikrosenberg) [inline HospitalGame_Business/src/business/BusinessHighScore.java]: It returns true if the player got added

File: HospitalGame_Business/src/business/BusinessHighScore.java [hunk 9]
```diff
@@ -10,67 +10,120 @@
...
             return true;
         }
         return false;
     }
-    
+
+    /**
+     * loops the entryset through the scores
+     *
+     * @return mangler at kommenteres
```
Comment (frederikrosenberg) [inline HospitalGame_Business/src/business/BusinessHighScore.java]: Says it all

File: HospitalGame_Business/src/business/Coordinate.java [hunk 10]
```diff
@@ -69,7 +73,11 @@ public static Coordinate add(Coordinate a, Coordinate b) {
     public static Coordinate subtract(Coordinate a, Coordinate b) {
         return new Coordinate(a.getX() - b.getX(), a.getY() - b.getY());
     }
-
+/**
+ * checks if object is a coordinate. 
+ * @param obj is an object
+ * @returns coordinates to object
+ */
```
Comment (frederikrosenberg) [inline HospitalGame_Business/src/business/Coordinate.java]: indent please

File: HospitalGame_Business/src/business/Coordinate.java [hunk 11]
```diff
@@ -78,6 +86,10 @@ public boolean equals(Object obj) {
         }
         return false;
     }
+    /**
+     * generates a random hashcode
+     * @return a random hashcode
+     */
 
     @Override
     public int hashCode() {
```
Comment (frederikrosenberg) [inline HospitalGame_Business/src/business/Coordinate.java]: Javadoc missing

File: HospitalGame_Business/src/business/Item/BloodBag.java [hunk 12]
```diff
@@ -33,7 +36,7 @@ public BloodBag(int bonusPoints, ItemName name, int weight, BloodType bloodType)
         super(bonusPoints, name, weight);
         this.bloodType = bloodType;
     }
-    
+
     public BloodBag(IBloodBag bloodBag) {
```
Comment (frederikrosenberg) [inline HospitalGame_Business/src/business/Item/BloodBag.java]: Javadoc missing

File: HospitalGame_Business/src/business/Item/IDCard.java [hunk 13]
```diff
@@ -7,10 +7,19 @@
...
+    * @author Robert Francisti
+    */
+
+    /**
+    * Constructor for IDcard, sets the weight and name
+    * @author rober
+    */
 public class IDCard extends Item {
 
     public IDCard(int weight, ItemName name) {
```
Comment (frederikrosenberg) [inline HospitalGame_Business/src/business/Item/IDCard.java]: Javadoc missing

File: HospitalGame_Business/src/business/Item/PowerUpItem.java [hunk 14]
```diff
@@ -5,14 +5,17 @@
...
- */
+    /**
+    * Class to handle all powerupitems in the game
+    *
+    * @author Frederik Schultz Rosenberg
+    * @author Andreas Bøgh Mølgaard-Andersen
+    * @author Lars Bjerregaard Jørgensen
+    * @author Robert Francisti
+    */
 public class PowerUpItem extends Item implements IPowerUpItem {
```
Comment (frederikrosenberg) [inline HospitalGame_Business/src/business/Item/PowerUpItem.java:19]: Javadoc missing some place in this class

File: HospitalGame_Business/src/business/Map.java [hunk 15]
```diff
@@ -146,24 +144,24 @@ public Room generateMap(int roomCount, List<IItem> items, List<INPC> npcs) {
                 break;
             }
         }
-        
+
         // Add every item to a random room.
         locked.setLocked(true);
-        
+        // Shuffles items in collections
```
Comment (frederikrosenberg) [inline HospitalGame_Business/src/business/Map.java]: in what collection?

File: HospitalGame_Business/src/business/Map.java [hunk 16]
```diff
@@ -228,7 +226,7 @@ public Room generateMap(int roomCount, List<IItem> items, List<INPC> npcs) {
             Room r = (Room) startRoom.getExit(key);
             queue.add(r);
         }
-
+        //sets room to startRoom
```
Comment (frederikrosenberg) [inline HospitalGame_Business/src/business/Map.java]: No. It adds the start room to the list

File: HospitalGame_Business/src/business/NPC/MoveAI.java [hunk 17]
```diff
@@ -11,10 +11,14 @@
...
+    * Class to handle the moveAI
+    *
+    * @author Frederik Schultz Rosenberg
+    * @author Andreas Bøgh Mølgaard-Andersen
+    * @author Lars Bjerregaard Jørgensen
+    * @author Robert Francisti
+    */
 public class MoveAI {
     
     private final int min = 5 * 1000;
```
Comment (frederikrosenberg) [inline HospitalGame_Business/src/business/NPC/MoveAI.java]: Javadoc missing

File: HospitalGame_Business/src/business/NPC/NPC.java [hunk 18]
```diff
@@ -42,15 +46,14 @@
...
-     * a reference to map
-     * Protected since porter need to call map.pathFinder.
+     * a reference to map Protected since porter need to call map.pathFinder.
      */
     protected Map map;
-    
+
     /**
-     * 
+     *
```
Comment (frederikrosenberg) [inline HospitalGame_Business/src/business/NPC/NPC.java]: Javadoc missing

File: HospitalGame_Business/src/business/NPC/NPC.java [hunk 19]
```diff
@@ -122,7 +135,7 @@ public String getDescription() {
     public IRoom getCurrentRoom() {
         return map.getRoomByID(currentRoomID);
     }
-    
+
     @Override
     public int getCurrentRoomID() {
```
Comment (frederikrosenberg) [inline HospitalGame_Business/src/business/NPC/NPC.java]: Javadoc missing

File: HospitalGame_Business/src/business/NPC/NPCFacade.java [hunk 20]
```diff
@@ -18,56 +16,86 @@
...
+ * @author Robert Francisti
  */
 public class NPCFacade implements INPCFacade {
 
     private Map map;
     private BusinessFacade business;
-
     private MoveAI moveAI;
-
     private ArrayList<NPC> NPCs = new ArrayList<>();
```
Comment (frederikrosenberg) [inline HospitalGame_Business/src/business/NPC/NPCFacade.java]: Javadoc missing

File: HospitalGame_Business/src/business/common/IItemFacade.java [hunk 21]
```diff
@@ -9,27 +9,94 @@
...
     public boolean activateItem(IItem item, int inventoryID, long startTime);
-
+    
+    /**
+     * updates the timeLeftOfBuff of a powerUpItem
+     *
+     * @param powerUpItem is the item being updated
+     * @param inventoryID
+     * @param lastUpdate
+     * @return
```
Comment (frederikrosenberg) [inline HospitalGame_Business/src/business/common/IItemFacade.java]: not filled out

**Review by frederikrosenberg — CHANGES_REQUESTED** [review]
There is a lot that is missing. 
Also please fix so there is no space between the java and the header of the function.
And the indentation of some of the Javadoc is also not done correctly.

Not sure i found all the places that are missing Javadoc, please check every class!

File: HospitalGame_Business/src/business/BusinessHighScore.java (hunk 9 above)
Comment (frederikrosenberg) [reply HospitalGame_Business/src/business/BusinessHighScore.java]:
```java
/**
 * Get the key for the score that has the lowest score
 *
 * @return the name/key with the lowest score
 */
```

**Review by frederikrosenberg — CHANGES_REQUESTED** [review]

**Review by frederikrosenberg — APPROVED** [review]

**Review by larsbjn — APPROVED** [review]

**Pull request merged**